Vates/VatesAPI/test/vtkDataSetFactoryTest.h

```diff
@@ -53,28 +53,35 @@ class vtkDataSetFactoryTest : public CxxTest::TestSuite
   };
 
 public:
-
-  void testSetSuccessor()
-  {
+  void testSetSuccessor() {
     MockvtkDataSetFactory factory;
-    MockvtkDataSetFactory* pSuccessor = new MockvtkDataSetFactory;
-    
-    EXPECT_CALL(factory, getFactoryTypeName()).WillOnce(testing::Return("TypeA")); 
-    EXPECT_CALL(*pSuccessor, getFactoryTypeName()).WillOnce(testing::Return("TypeB")); //Different type name, so setting the successor should work.
-    factory.SetSuccessor(pSuccessor);
+    auto successor = Mantid::Kernel::make_unique<MockvtkDataSetFactory>();
+    MockvtkDataSetFactory *pSuccessor = successor.get();
```

Comment:
`pSuccessor` seems to be used only once? Defining it was more confusing than helpful for me.